pkg/dependencies/helmfactory.go

```diff
@@ -0,0 +1,138 @@
+package dependencies
+
+import (
+	"context"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// ExecutableBuilder builds the Helm executeble and returns it.
+type ExecutableBuilder interface {
+	BuildHelmExecutable(...executables.HelmOpt) *executables.Helm
+}
+
+// HelmFactoryOpt configures a HelmFactory.
+type HelmFactoryOpt func(*HelmFactory)
+
+// HelmFactory is responsible for creating and owning instances of Helm client.
+type HelmFactory struct {
+	client         cluster.Client
+	helmClient     *cluster.HelmClient
+	mu             sync.Mutex
+	builder        ExecutableBuilder
+	registryMirror *registrymirror.RegistryMirror
+	env            map[string]string
+	insecure       bool
+}
+
+// WithRegistryMirror configures the factory to use registry mirror wherever applicable.
+func WithRegistryMirror(registryMirror *registrymirror.RegistryMirror) HelmFactoryOpt {
+	return func(hf *HelmFactory) {
+		hf.registryMirror = registryMirror
+	}
+}
+
+// WithEnv configures the factory to use proxy configurations wherever applicable.
+func WithEnv(env map[string]string) HelmFactoryOpt {
+	return func(hf *HelmFactory) {
+		hf.env = env
+	}
+}
+
+// WithInsecure configures the factory to configure helm to use to allow connections to TLS registry without certs or with self-signed certs.
+func WithInsecure() HelmFactoryOpt {
+	return func(hf *HelmFactory) {
+		hf.insecure = true
+	}
+}
+
+// NewHelmFactory returns a new HelmFactory.
+func NewHelmFactory(client cluster.Client, builder ExecutableBuilder, opts ...HelmFactoryOpt) *HelmFactory {
+	hf := &HelmFactory{
+		client:  client,
+		builder: builder,
+		mu:      sync.Mutex{},
+	}
+
+	for _, o := range opts {
+		o(hf)
+	}
+
+	return hf
+}
+
+// GetClient returns a new Helm executeble.
+func (f *HelmFactory) GetClient(opts ...executables.HelmOpt) cluster.Helm {
```

Comment:
What is this method for?

Reply:
This is actually suppose to be a helper that builds and returns the `Helm` executeable object. I had forgotten to re-adjust this after some changes. I plan to change this `buildClient`

Reply:
got it, let's juts make it unexported then

Reply:
Done